components/brave_wallet/browser/solana_tx_manager.cc

```diff
@@ -27,10 +28,61 @@
 #include "brave/components/brave_wallet/common/brave_wallet.mojom.h"
 #include "brave/components/brave_wallet/common/brave_wallet_constants.h"
 #include "brave/components/brave_wallet/common/brave_wallet_types.h"
+#include "brave/components/brave_wallet/common/encoding_utils.h"
+#include "brave/components/brave_wallet/common/solana_address.h"
 #include "brave/components/brave_wallet/common/solana_utils.h"
+#include "build/build_config.h"
 #include "components/grit/brave_components_strings.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
 #include "ui/base/l10n/l10n_util.h"
 
+namespace {
+
+std::optional<uint8_t> DecodeUint8(const std::vector<uint8_t>& input,
+                                   size_t& offset) {
+  if (offset >= input.size() || input.size() - offset < sizeof(uint8_t)) {
+    return std::nullopt;
+  }
+
+  offset += sizeof(uint8_t);
+  return input[offset - sizeof(uint8_t)];
+}
+
+std::optional<uint32_t> DecodeUint32(const std::vector<uint8_t>& input,
+                                     size_t& offset) {
+  if (offset >= input.size() || input.size() - offset < sizeof(uint32_t)) {
+    return std::nullopt;
+  }
+
+  // Read bytes in little endian order.
+  base::span<const uint8_t> s =
+      base::make_span(input.begin() + offset, sizeof(uint32_t));
+  uint32_t uint32_le = *reinterpret_cast<const uint32_t*>(s.data());
+
+  offset += sizeof(uint32_t);
+
+#if defined(ARCH_CPU_LITTLE_ENDIAN)
+  return uint32_le;
+#else
+  return base::ByteSwap(uint32_le);
+#endif
+}
```

Comment:
Chromium is LE only
https://source.chromium.org/chromium/chromium/src/+/main:base/numerics/byte_conversions.h;l=19

Other cases where we try to support both don't make sense also